### ISSUE TITLE
The topic once deleted and recreated not listed on the /topic endpoint

### ISSUE THREAD

**Issue by PlkMarudny** [nsqio/nsq#1009]
I have a simple Docker setup, one nsqd and one nsqlookupd.

1. I create a topic using a /pub endpoint on a nsqd
2. I check the topic list using /topics on nsqlookupd - the topic is created
3. I delete a topic using /topic/delete?topic= endpoint on nslookupd
4. I check like in point 2. - the topic was deleted
5. After 2 minutes I try to repeat point 1. - receving OK 200, but trying to check the list of topics does not display the (re)created one
6. I restart the nsqd daemon using `docker restart nsqd`
7. The (re)created topic appears on the list of topics using /topics endpoint

So basically the topic once used and recreated is not visible on the /topic endpoint. Topics deleted using nsaqadmin behave correctly, i.e. the effect described above does not apply.

What is the correct way of deleting a topic then?

**Comment by ploxiln** [comment]
This is probably due to "tombstones": http://nsq.io/components/nsqlookupd.html#deletion-and-tombstones

It is not expected that a topic will be deleted and then soon recreated. It is expected that you would only delete a topic, on all servers or on a particular server, if that topic should not exist there for the foreseeable future. Otherwise, you could stop publishing to the topic, and you could empty the topic and the topic's channels.

**Comment by PlkMarudny** [comment]
The whole process is rather unclear. Deleting from the nsqlookupd does not remove the topic from nsqd as I am observing now (checked via /stats on a node). So publishing to it again (i have only one node) gives 200 OK. But the topic does not want to appear back in nsqlookupd. Cannot see any tombstoning involved (I guess this hould be done by nsqlookupd).

But if nslookupd /topic/delete does not really delete, isn't this a potential resource leak?

In other words, how to delete a topic? I have topics that I do not want to keep (used for progress report of some tasks). They are short-lived. But they can eventually reappear. I can generate random names, this would solve the reuse issue, but what about the deletion?

**Comment by PlkMarudny** [comment]
After playing more with HTTP API, it looks like:

1. nsqlookup /topic/delete only removes the topic from the metadata, no tombstoning done
2. to get rid of a topic, it should be removed from metadata and than from each node

**Comment by ploxiln** [comment]
Yup. nsqadmin can do this for you. It ends up calling this function:
https://github.com/nsqio/nsq/blob/v1.0.0-compat/internal/clusterinfo/data.go#L712-L751

**Comment by mreiferson** [comment]
As @ploxiln mentioned, the easiest way to handle these situations is by using `nsqadmin`, which Does The Right Thing ™️ for both deletion and tombstoning.
